### PR TITLE
Add misk.jdbc.Transacter, create misk-jdbc-testing

### PR DESCRIPTION
**Pull request by mightyguava**
I started adding a very basic `misk.jdbc.Transacter` but realized that to test it properly, I need to split `TruncateTablesService` out of `misk-hibernate-testing`. And so one thing led to another and this turned into a nightmare...

But the tests pass and the change is theoretically backwards compatible.

* `misk-hibernate-testing` is now just a placeholder with a `HibernateTestingModule` that installs `JdbcTestingModule`, everything's been moved to `misk-jdbc-testing`.
* All dependencies on `misk.hibernate` have been removed from `misk-jdbc-testing`. This was a pretty monumental effort because from the code perspective, the Transacter, Vitess, Hibernate, are tightly coupled together.
* The hardest part was splitting out the `{Vitess,MySQ}ScaleSafetyChecks`, as they were really tightly interwoven with Transacter, Session, and Connection. Could really use a second set of eyes there.

### CODE REVIEW

File: misk-jdbc/src/main/kotlin/misk/jdbc/SchemaMigrator.kt [hunk 1]
```diff
@@ -104,10 +98,12 @@ internal class SchemaMigrator(
   private val qualifier: KClass<out Annotation>,
   private val resourceLoader: ResourceLoader,
   private val dataSourceConfig: DataSourceConfig,
-  private val dataSource: Provider<DataSource>,
+  private val dataSource: DataSourceService,
   private val connector: DataSourceConnector
 ) {
 
+  val shards = misk.vitess.shards(dataSource)
```
Comment (tirsen) [inline misk-jdbc/src/main/kotlin/misk/jdbc/SchemaMigrator.kt:105]: Ah right, this will cache properly

**Review by tirsen — APPROVED** [review]
Coooool!

File: misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt [hunk 2]
```diff
@@ -17,6 +19,10 @@ class VitessDialect : MySQL57Dialect() {
         throw CowriteException(message, sqlException)
       }
 
+      if (sqlException is TableScanException || sqlException is CowriteException || sqlException is FullScatterException) {
```
[the remark this answers is not in the document]
Comment (mightyguava) [reply misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt:22]: That is a good idea!

File: misk-hibernate/src/test/kotlin/misk/hibernate/VitessScaleSafetyTest.kt [hunk 3]
```diff
@@ -151,7 +153,7 @@ class VitessScaleSafetyTest {
 
   @Test
   fun crossShardQueriesAreDetected() {
-    assertThrows<FullScatterException> {
+    assertThrows<UndeclaredThrowableException> {
```
[the remark this answers is not in the document]
Comment (mightyguava) [reply misk-hibernate/src/test/kotlin/misk/hibernate/VitessScaleSafetyTest.kt:156]: Dude, I tried. For hours. HOURS. To fix this ScaleSafetyTest. No. Maybe someone who is more familiar with Hibernate and JDBC's exception mapping fiasco can figure it out

File: misk-jdbc/src/main/kotlin/misk/vitess/Shards.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package misk.vitess
+
+import com.google.common.base.Suppliers
+import misk.jdbc.DataSourceService
+import misk.jdbc.map
+import java.sql.SQLRecoverableException
+import java.util.concurrent.TimeUnit
+import java.util.function.Supplier
+
+fun shards(dataSourceService: DataSourceService): Supplier<Set<Shard>> =
```
[the remark this answers is not in the document]
Comment (mightyguava) [reply misk-jdbc/src/main/kotlin/misk/vitess/Shards.kt:10]: Probably. But I expect this to only ever be used by platform-code for Vitess which is on its way out.

File: misk-jdbc/src/main/kotlin/misk/jdbc/Check.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.util.EnumSet
+
+enum class Check {
+  FULL_SCATTER,
+  TABLE_SCAN,
+  COWRITE
+}
+
+object CheckDisabler {
```
Comment (ChunYin) [inline misk-jdbc/src/main/kotlin/misk/jdbc/Check.kt:11]: I liked this refactoring here.

**Review by ChunYin — APPROVED** [review]
This is a massive change in one of the misk libraries. How do you plan to test this before it gets rolled out to all misk services?

File: misk-jdbc/src/main/kotlin/misk/jdbc/Check.kt (hunk 5 above)
Comment (mightyguava) [reply misk-jdbc/src/main/kotlin/misk/jdbc/Check.kt:11]: > This is a massive change in one of the misk libraries. How do you plan to test this before it gets rolled out to all misk services?

I tried running newswriter tests. They all pass except when merging is involved due to some weird classpath bug that I think is unrelated. So we should be all good!

**Pull request merged**